File: Demo/MNG-Ads-SDK/InterstitialManager.m
//
//  InterstitialManager.m
//  MNG-Ads-SDK-Demo
//
//  Created by Ben Salah Med Amine on 4/23/15.
//  Copyright (c) 2015 MNG. All rights reserved.
//

#import "InterstitialManager.h"

static InterstitialManager * _sharedManager;

@implementation InterstitialManager{
    MNGAdsSDKFactory *interstitialAdsFactory;
}

+(InterstitialManager*)sharedManager{
    if (!_sharedManager) {
        _sharedManager = [[InterstitialManager alloc]init];
    }
    return _sharedManager;
}

-(void)showInterstitial{
    if ([NSUserDefaults.standardUserDefaults objectForKey:@"IABTCF_TCString"] == nil) {
        NSLog(@"IABTCF_TCString not yet allowed");
        return;
    }
    if (interstitialAdsFactory.isBusy) {
        NSLog(@"Ads Factory is busy");
        return;
    }
    [interstitialAdsFactory releaseMemory];
    interstitialAdsFactory = [[MNGAdsSDKFactory alloc]init];
    interstitialAdsFactory.interstitialDelegate = self;
    interstitialAdsFactory.clickDelegate = self;
    interstitialAdsFactory.viewController = [APP_DELEGATE drawerViewController];
    interstitialAdsFactory.placementId = MNG_ADS_INTERSTITIAL_PLACEMENT_ID;
    /**
     * MNGPrefence is the same for all ad types
     * @important: your application can be rejected by Apple if you use the device's location only for advertising
     **/
    MNGPreference *preferences = [Utils getTestPreferences] ;
    [interstitialAdsFactory loadInterstitialWithPreferences:preferences autoDisplayed:NO];
}

-(void)showInterstitialOverlay{
    if (interstitialAdsFactory.isBusy) {
        NSLog(@"Ads Factory is busy");
        return;
    }
    if ([NSUserDefaults.standardUserDefaults objectForKey:@"IABTCF_TCString"] == nil) {
        NSLog(@"IABTCF_TCString not yet allowed");
        return;
    }
    [interstitialAdsFactory releaseMemory];
    interstitialAdsFactory = [[MNGAdsSDKFactory alloc]init];
    interstitialAdsFactory.interstitialDelegate = self;
    interstitialAdsFactory.clickDelegate = self;
    interstitialAdsFactory.viewController = [APP_DELEGATE drawerViewController];
    interstitialAdsFactory.placementId = MNG_ADS_INTERSTITIAL_OVERLAY_PLACEMENT_ID;
    /**
     * MNGPrefence is the same for all ad types
     * @important: your application can be rejected by Apple if you use the device's location only for advertising
     **/
    MNGPreference *preferences = [Utils getTestPreferences] ;
    [interstitialAdsFactory loadInterstitialWithPreferences:preferences];
}

#pragma mark - MNGAdsAdapterInterstitialDelegate

-(void)adsAdapterInterstitialDidLoad:(MNGAdsAdapter *)adsAdapter{
    NSLog(@"adsAdapterInterstitialDidLoad");
    [adsAdapter showAdFromRootViewController:[APP_DELEGATE drawerViewController] animated:NO] ;
}

-(void)adsAdapterInterstitialDisappear:(MNGAdsAdapter *)adsAdapter{
    NSLog(@"adsAdapterInterstitialDisappear");
}

-(void)adsAdapter:(MNGAdsAdapter *)adsAdapter interstitialDidFailWithError:(NSError *)error{
    NSLog(@"%@",error);
    
}
-(void)adsAdapterInterstitialDidShown:(MNGAdsAdapter *)adsAdapter{
    NSLog(@"adsAdapterInterstitialDidShown");

}
-(void)adsAdapterAdWasClicked:(MNGAdsAdapter *)adsAdapter{
    NSLog(@"adsAdapterAdWasClicked");
}

@end

File: DemoSwiftWithSPM/DemoSwift-SPM/DemoSwift-SPM/InterstitialManager(from background)/InterstitialManager.swift
//
//  InterstitialManager.swift
//  BlueStackDemoDev
//
//  Created by MedSghaier on 22/8/2022.
//

import Foundation
import BlueStackSDK

class InterstitialManager : NSObject , MNGAdsAdapterInterstitialDelegate , MNGClickDelegate  {
    
    //MARK: -Singleton
    static let _sharedManager = InterstitialManager()
    
    var interstitialAdsFactory: MNGAdsSDKFactory?
    
    private override init(){
        
    }
    
    //MARK: - Preferences
    func getTestPreferences() ->MNGPreference {
        let preference = MNGPreference()
        preference.keyWord = "target=mngadsdemo;version=3.0.4;semantic=https://www.google.com"
        preference.gender = MNGGender.male
        preference.setLocationPreferences(CLLocation.init(latitude: 48.87610, longitude: 10.453), withConsentFlag: 2)
        preference.contentUrl = "your content url"
        preference.age = 60
        return preference
    }
    
    //MARK: - Load Interstitials
    
    func showInterstitial(){
        
        if interstitialAdsFactory?.isBusy ?? false {
            NSLog("Ads Factory is busy")
            return
        }
        
        interstitialAdsFactory?.releaseMemory()
        interstitialAdsFactory = MNGAdsSDKFactory()
        interstitialAdsFactory?.interstitialDelegate = self
        interstitialAdsFactory?.clickDelegate = self
        interstitialAdsFactory?.viewController = APP_DELEGATE.drawerViewController
        interstitialAdsFactory?.placementId = "/\(DemoSwiftConstants.appID)/interstitial"
        let preferences = self.getTestPreferences()
        interstitialAdsFactory?.loadInterstitial(withPreferences: preferences, autoDisplayed: false)
    }
    
    func showInterstitialOverlay(){
        
        if interstitialAdsFactory?.isBusy ?? false {
            NSLog("Ads Factory is busy")
            return
        }
        
        interstitialAdsFactory?.releaseMemory()
        interstitialAdsFactory = MNGAdsSDKFactory()
        interstitialAdsFactory?.interstitialDelegate = self
        interstitialAdsFactory?.clickDelegate = self
        interstitialAdsFactory?.viewController = APP_DELEGATE.drawerViewController
        interstitialAdsFactory?.placementId = "/\(DemoSwiftConstants.appID)/interstitialOverlay"
        let preferences = self.getTestPreferences()
        interstitialAdsFactory?.loadInterstitial(withPreferences: preferences)
    }
    
    //MARK: -interstitial Delegate
    func adsAdapterInterstitialDidLoad(_ adsAdapter: MNGAdsAdapter!) {
        NSLog("adsAdapterInterstitialDidLoad");
        adsAdapter.showAd(fromRootViewController: APP_DELEGATE.drawerViewController, animated: false)
    }
    
    func adsAdapter(_ adsAdapter: MNGAdsAdapter!, interstitialDidFailWithError error: Error!) {
        NSLog("\(String(describing: error))")
    }
    
    func adsAdapterInterstitialDisappear(_ adsAdapter: MNGAdsAdapter!) {
        NSLog("adsAdapterInterstitialDisappear");
    }
    
    func adsAdapterInterstitialDidShown(_ adsAdapter: MNGAdsAdapter!) {
        NSLog("adsAdapterInterstitialDidShown");
    }
    
    func adsAdapterAdWasClicked(_ adsAdapter: MNGAdsAdapter!) {
        NSLog("adsAdapterAdWasClicked");
    }
    
}
